includes/classes/usu.php

```diff
@@ -625,26 +605,19 @@ protected function get_product_name($pID, $cID = null)
                 break;
         }
 
-        // Add the category
-        if (USU_CATEGORY_DIR != 'off') {
-            $sql = 
-                "SELECT ptc.categories_id AS c_id, p.master_categories_id AS master_id
-                   FROM " . TABLE_PRODUCTS . " p
-                        LEFT JOIN " . TABLE_PRODUCTS_TO_CATEGORIES . " AS ptc
-                            ON ptc.products_id = p.products_id
-                  WHERE p.products_id = $pID" . $categories_clause . "
-                  LIMIT 1";
-
-
-            $result = $db->Execute($sql, false, true, 43200);
-            $category = '';
-            if (!$result->EOF) {
-                $cID = ($cID !== null ? $cID : (int)$result->fields['master_id']);
-                $category = $this->get_category_name($cID) . $this->reg_anchors['cPath'] . $cID . '/';
-            }
-
-            $return = $category . $return;
-        }
+	    // Add the category
+	    $category = '';
+	    if (USU_CATEGORY_DIR != 'off') {
+		    if (empty($cID)) {
+			    $masterCatID = (int)zen_get_products_category_id($pID);
+			    $category = $this->get_category_name($cID) . $this->reg_anchors['cPath'] . $masterCatID . '/';
+		    } else {
+			    if (zen_product_in_category($pID, $cID)) {
+				    $category = $this->get_category_name($cID) . $this->reg_anchors['cPath'] . $cID . '/';
+			    }
+		    }
+		    $return = $category . $return;
+	    }
```

Comment:
this section of code does the same thing as the categories clause.  unfortunately i am now seeing that `zen_product_in_category` does not work on the admin side, so we would need to redefine it; perhaps in the extra_functions?  i only noticed this problem this morning as searching for products.  it does successfully do both parent and original USU_FORMAT.